packages/react-components/react-carousel-preview/library/src/components/CarouselNavButton/useCarouselNavButton.ts

```diff
@@ -46,11 +54,25 @@ export const useCarouselNavButton_unstable = (
         ref: ref as React.Ref<HTMLButtonElement>,
         role: 'tab',
         type: 'button',
+        'aria-selected': selected,
+        'aria-controls': controlledSlides,
         ...defaultTabProps,
       },
     },
   );
 
+  useIsomorphicLayoutEffect(() => {
+    return subscribeForValues(data => {
+      const controlList = data.groupIndexList[index];
+      const _controlledSlideIds = controlList
+        .map((slideIndex: number) => {
+          return data.slideNodes[slideIndex].id;
+        })
+        .join(' ');
+      setControlledSlides(_controlledSlideIds);
```

Comment:
This will cause double render (render + re-render for `controlledSlides`). Can we do:

```
if (buttonRef.current) {
  buttonRef.current.ariaControls = _controlledSlideIds;
}
```

It will be impossible for consumer to customize the attr, but we will avoid a re-render